draft-dnsop-deleg.md

```diff
@@ -315,14 +316,15 @@ If a DELEG RRset is present at the delegation point, the name server MUST return
 
 If no DELEG RRset is present at the delegation point, and the zone was signed with a DNSKEY that has the DELEG flag set, the name server MUST return the NSEC or NSEC3 RR that proves that the DELEG RRset is not present including its associated RRSIG RR along with the DS RRset and its associated RRSIG RR if present and the NS RRset, if present. 
 
-Including these DELEG, DS, NSEC or NSEC3, and RRSIG RRs increases the size of referral messages. If space does not permit inclusion of these records, including glue address records, the name server MUST set the TC bit.
+Including these DELEG, DS, NSEC or NSEC3, and RRSIG RRs increases the size of referral messages. If space does not permit inclusion of these records, including glue address records, the name server MUST set the TC bit on the response.
 
 ### Responding to Queries for Type DELEG
-DELEG records, when present, are included in referrals. When a parent and child are served from the same authoritative server, this referral will not be send since the authoritative server will respond with information from the child zone. In that case the resolver may query for type DELEG.
 
-The DELEG resource record type is unusual in that it appears only on the parent zone's side of a zone cut.  For example, the DELEG RRset for the delegation of "foo.example" is stored in the "example" zone rather than in the "foo.example" zone.  This requires special processing rules for both name servers and resolvers, as the name server for the child zone is authoritative for the name at the zone cut by the normal DNS rules but the child zone does not contain the DELEG RRset.
+DELEG records, when present, are included in referrals. When a parent and child are served from the same authoritative server, this referral will not be sent because the authoritative server will respond with information from the child zone. In that case, the resolver may query for type DELEG.
+
```

Comment:
I think this section shows the hidden complexity of the proposed "DELEG+NS" referral response, and we shouldn't be too hasty in concluding that it is the way forward.  For -00, I think it's fine but we could use some additional disclaimers about the possible downsides.

```suggestion

FOR DISCUSSION: The additional complexity here, with resolvers needing two very different resolution behaviors, merits further investigation.  It also opens some potential paradoxes.  For example, if the server is authoritative for the child via DELEG, but not via NS, then it cannot serve any answer that will work for both DELEG-aware and non-DELEG-aware resolvers.
```

Reply:
Fair point. I also wonder how often of a case this would be and whether or not it is worth the special handling at all.

Reply:
Moin!

uk and co.uk
au and com.au

comes to my mind immediately. It is similar then DS which has the exact same kind of handling.

So long
-Ralf

Reply:
Since this PR was closed, I opened #33 to discuss the second part of this issue.

Reply:
ok, so I suppose this could help with sharding migration of individual second-level domains.

As much as it is probably not required in that case, there may be implications for  `draft-dnsop-deleg-dnssec.md` though.